src/jit/ee_il_dll.cpp

```diff
@@ -163,6 +163,20 @@ CorJitResult CILJit::compileMethod (
         return g_realJitCompiler->compileMethod(compHnd, methodInfo, flags, entryAddress, nativeSizeOfCode);
     }
 
+    if (!JitConfig.isInitialized())
```

Comment:
> There's an interesting assumption here that any compHnd can be used to initialize JitConfig, exactly once. Subsequent calls to compileMethod() don't need to reinitialize, and we assume that every ICorJitInfo would have returned the same data. This is probably true for this case, but maybe a note needs to be made (in the interface header?) about this assumption.

This is something that Russ and I discussed as well. Unfortunately, the assumption that configuration values are ambiently available and constant over the lifetime of the jit is pretty firmly baked in at the moment (e.g. config values are cached and there is code that uses config values that doesn't have access to an `ICorJitInfo` instance that could be used to fetch updated values). This definitely deserves a note, especially since it would be a very difficult assumption to remove.